Clarify managed identities doc (#4024)
dapr/docs: daprdocs/content/en/developing-applications/integrations/Azure/azure-authentication/howto-mi.md

Using Managed Identities (MI), authentication happens automatically by virtue of your application running on top of an Azure service that has an assigned identity. 

Let's say you enable a managed service identity using an Azure KeyVault secrets component for an Azure service. When you do, an Microsoft Entra ID application is created for you and automatically assigned to the service. Your Dapr services can then leverage that identity to authenticate with Microsoft Entra ID, transparently and without you having to specify any credentials.

> Dapr supports both system-assigned and user-assigned identities.

> In your component YAML, the [`azureClientId` property]({{< ref "authenticating-azure.md#authenticating-with-managed-identities-mi" >}}) only needs to be set if using user-assigned identity. Otherwise, you can omit this property for system-managed identity to be used by default.

Reviewer: ```suggestion
In your component YAML, the [`azureClientId` property]({{< ref "authenticating-azure.md#authenticating-with-managed-identities-mi" >}}) only needs to be set if using user-assigned identity. Otherwise this property can be omitted and system-managed identity will be used by default.
```

## System-assigned

If you set up system-assigned MI using an Azure KeyVault component, the YAML would look like the following:

```yml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: azurekeyvault
spec:
  type: secretstores.azure.keyvault
  version: v1
  metadata:
  - name: vaultName
    value: mykeyvault
```

In this example, the system-assigned MI looks up the service identity and communicates with the `mykeyvault` vault.

## User-assigned

> If you set up user-assigned MI using an Azure KeyVault component, the YAML would look like the following:

Reviewer: I would keep managed ONLY for managed identities and otherwise simply say user assigned identities.
```suggestion
If you set up user-assigned identity using an Azure KeyVault component, the YAML would look like the following:
```

```yml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: azurekeyvault
spec:
  type: secretstores.azure.keyvault
  version: v1
  metadata:
  - name: vaultName
    value: mykeyvault
  - name: azureClientId
    value: someAzureIdentityClientIDHere
```

## Grant access to the service

> Once you've set up the component YAML with the `azureClientId` property, you need to perform the requisite Microsoft Entra ID role assignments to grant the system-managed or user-managed identity access to the desired service. 

Reviewer: This is confusing. The `azureClientId` property, is only needed for user assigned identities. If you set up a service managed identity then you do not need this property and the identity is simply passed to the component. Can you rephrase this?

> You can do this by assigning an identity to a new or existing Azure resource. The instructions depend on the service use. Check the following official documentation for the most appropriate instructions:

Reviewer: ```suggestion
You can set up a managed identity to a new or existing Azure resource. The instructions depend on the service use. Check the following official documentation for the most appropriate instructions:
```
